src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/IntrinsicSupport/EqualityComparerHelpers.cs

```diff
@@ -59,6 +59,10 @@ internal static object GetComparer(RuntimeTypeHandle t)
             RuntimeTypeHandle openComparerType = default(RuntimeTypeHandle);
             RuntimeTypeHandle comparerTypeArgument = default(RuntimeTypeHandle);
 
+            if (typeof(byte).TypeHandle.Equals(t))
+            {
+                return new ByteEqualityComparer();
```

Comment:
Does ByteEqualityComparer still provide any performance benefits? Would it be better to delete `ByteEqualityComparer` from here? (The type itself needs to stay for binary serialization compatibility.)

Reply:
It seems that it's not needed for `EqualityComparer` (codegen wise), but might be still needed for just `Comparer`

Reply:
![image](https://github.com/dotnet/runtime/assets/523221/d9462691-c8f5-44b8-a0e0-ccfa717c929e)

Reply:
> Does ByteEqualityComparer still provide any performance benefits? Would it be better to delete `ByteEqualityComparer` from here? (The type itself needs to stay for binary serialization compatibility.)

Not sure, but I just wanted to unify the behaviour so that all runtimes return the same thing here. I could remove it for all runtimes instead, just it'd be a slightly breaking change then I guess.



> It seems that it's not needed for `EqualityComparer` (codegen wise), but might be still needed for just `Comparer`

Only `EqualityComparer` is special-cased like this, `Comparer` isn't.

Reply:
> it'd be a slightly breaking change then I guess.

It should not be a breaking change per definition of a breaking change. Changing the internal type that is returned from a factory method is allowed change.

Reply:
`// Specialize for byte so Array.IndexOf is faster.`
Do you know what does this comment on `ByteEqualityComparer` refer to?

Reply:
It refers to optimized implementations of IndexOf methods on ByteEqualityComparer that no longer exists: https://github.com/dotnet/coreclr/blob/release/1.0.0/src/mscorlib/src/System/Collections/Generic/EqualityComparer.cs#L328-L365